c7n/resources/s3.py

```diff
@@ -3283,7 +3283,10 @@ def filter_bucket(self, b, sse):
         key = self.get_key(b)
         crypto = self.data.get('crypto')
         rule = sse.get('ApplyServerSideEncryptionByDefault')
-        algo = rule.get('SSEAlgorithm')
+        try:
```

Comment:
nit: we may be able to cover the problem case here a little more explicitly by using `if not algo` or `if algo is None` rather than catching an exception on `.get()`.

Reply:
That's a good point.  I suppose I could have done that too.  Would you like me to modify the PR?

Reply:
Yeah I think the tweak would accomplish the same thing (covering a non-existent default encryption algorithm) and be a bit clearer. I think that would address @kapilt's question too.

Reply:
I agree.  I'll submit the updated PR tomorrow.  As far as the traceback is concerned, there wasn't anything generated except for the following message for a resource experiencing the issue:

2022-07-31 00:11:45,319 - custodian.filters - ERROR - Message: 'NoneType' object has no attribute 'get' Bucket: xcc-services-alb-access-logs-prod-us-east-1

Thanks again @ajkerrigan